src/frontend/src/pages/GanttPage/GanttChartComponents/GanttTaskBar/GanttTaskBarView.tsx

```diff
@@ -34,156 +40,200 @@ const GanttTaskBarView = ({
+                  WebkitLineClamp: 1,
+                  userSelect: 'none'
                 }}
-                onMouseOver={handleOnMouseOver}
-                onMouseLeave={handleOnMouseLeave}
-                onClick={() => history.push(`${`${routes.PROJECTS}/${event.id}`}`)}
               />
-            );
-          })}
-        {highlightedChange && (
+            </Box>
+          </div>
           <div
-            id="proposedChange"
             style={{
-              paddingTop: '2px',
-              paddingLeft: '5px',
-              gridColumnStart:
-                highlightedChange.eventId === event.id
-                  ? days.findIndex((day) => dateToString(day) === dateToString(highlightedChange.newStart)) + 1
-                  : getStartCol(event.start),
-              gridColumnEnd:
-                highlightedChange.eventId === event.id
-                  ? days.findIndex((day) => dateToString(day) === dateToString(highlightedChange.newEnd)) === -1
-                    ? days.length + 1
-                    : days.findIndex((day) => dateToString(day) === dateToString(highlightedChange.newEnd)) + 2
-                  : getEndCol(event.end),
-              height: '2rem',
-              border: `1px solid ${theme.palette.text.primary}`,
-              borderRadius: '0.25rem',
-              backgroundColor: '#ef4345',
-              cursor: 'pointer',
               gridRow: 1,
-              zIndex: 6
+              zIndex: 3,
+              gridColumnStart: getStartCol(task.start),
+              gridColumnEnd: getEndCol(task.end),
+              display: 'flex',
+              alignItems: 'center',
+              marginTop: isProject ? '-10px' : undefined,
+              marginBottom: isProject ? '-10px' : undefined,
+              cursor: 'pointer',
+              width: isProject ? 'fit-content' : '100%'
             }}
+            onMouseOver={handleOnMouseOver}
+            onMouseLeave={handleOnMouseLeave}
+            onClick={!isProject ? () => history.push(`${`${routes.PROJECTS}/${task.id}`}`) : undefined}
           >
+            {isProject && (
+              <IconButton onClick={onWorkPackageToggle} sx={{ marginRight: '-15px', marginLeft: '-5px' }}>
+                {showWorkPackages ? <ArrowDropDownIcon fontSize="large" /> : <ArrowRightIcon fontSize="large" />}
+              </IconButton>
+            )}
             <Typography
               variant="body1"
               sx={{
-                color: event.styles ? event.styles.color : '#ffffff',
+                color: task.styles ? task.styles.color : '#ffffff',
+                px: 1,
                 overflow: 'hidden',
                 textOverflow: 'ellipsis',
                 whiteSpace: 'nowrap'
               }}
+              onClick={onWorkPackageToggle}
             >
-              {event.name}
+              {task.name}
             </Typography>
           </div>
-        )}
+          {!highlightedChange &&
+            isProject &&
```

Comment:
make a variable for the combo of these boolean checks that makes it easy to understand what you're checking here